tests: stackprot: don't set main stack size

tyrantlizard77 opened the pull request:
This test uses ztest anyway, the default should be fine
just like any other test running under ztest.

Signed-off-by: Andrew Boie <andrew.p.boie@intel.com>

Commit:
tests: stackprot: don't set main stack size

This test uses ztest anyway, the default should be fine
just like any other test running under ztest.

k_thread_create() uses a lot of stack, and the main
stack size is very small if ztest is enabled. Do it in
another ztest task instead.

We don't need to mess with the main thread's priority,
just have the alt thread run cooperatively.

Signed-off-by: Andrew Boie <andrew.p.boie@intel.com>

andrewboie commented:
CI issues should be fixed

andrewboie commented:
Something very strange is going on here with Riscv32
@nategraff-sifive any idea?

Commit:
kernel: adjust default main stack for riscv

We are just at the knife edge with 512, with stack
overflows being observed with stack canaries enabled.
Given the special case for the idle thread stack size
on this arch, seems reasonable to increase it here
for that arch.

Signed-off-by: Andrew Boie <andrew.p.boie@intel.com>
